core/Symbols.cc

```diff
@@ -30,8 +30,9 @@ vector<TypePtr> Symbol::selfTypeArgs(const GlobalState &gs) const {
     ENFORCE(isClass()); // should be removed when we have generic methods
     vector<TypePtr> targs;
     for (auto tm : typeMembers()) {
-        if (tm.data(gs)->isFixed()) {
-            targs.emplace_back(tm.data(gs)->resultType);
+        auto *lambdaParam = cast_type<LambdaParam>(tm.data(gs)->resultType.get());
+        if (lambdaParam != nullptr && lambdaParam->isFixed()) {
+            targs.emplace_back(lambdaParam->lower);
```

Comment:
why lower rather than upper?

Reply:
Since the type variable is fixed, `lower` and `upper` will be the same. I picked `lower` arbitrarily, but `upper` would be fine too.